src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,98 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.DukeException;
+import duke.parser.DateParser;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Reads the given file to get the stored tasks and store ongoing tasks after exiting duke.
+ */
+public class Storage {
+    private ArrayList<Task> storedTasks;
+    private String filePath = "./data/sheep.txt";
+
+    /**
+     * Constructor for <code>Storage</code>.
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Read the tasks from a given file and return a task list with those tasks.
+     * @return An ArrayList of Task
+     * @throws DukeException
+     */
+    public ArrayList<Task> load() throws DukeException {
+        File file = new File(filePath);
+        try {
+            Scanner sc = new Scanner(file);
+            ArrayList<Task> tasks = new ArrayList<>();
+            while (sc.hasNext()) {
+                String storedTask = sc.nextLine();
+                String[] splitTask = storedTask.split(" \\| ");
+                Task task;
+                if (splitTask[0].equals("T")) {
+                    task = new Todo(splitTask[2]);
+                } else if (splitTask[0].equals("D")) {
+                    if (DateParser.isDateValid(splitTask[2])) {
+                        task = new Deadline(splitTask[2], DateParser.convertToLocalDate(splitTask[2]));
+                    } else {
+                        System.out.println("OOPS!!! Look like your date format is incorrect");
+                        continue;
+                    }
+                } else {
+                    task = new Event(splitTask[2], splitTask[3]);
+                }
+                if (splitTask[1].equals("1")) {
+                    task.markDone();
+                }
+                tasks.add(task);
+            }
+            return tasks;
+        } catch (FileNotFoundException e) {
```

Comment:
Avoid Deep Nesting
It would be better to reduce the level of indentation to not more than 3.